index.js

```diff
@@ -1,9 +1,10 @@
 'use strict';
 module.exports = function (str) {
+	function escapeIfNotPrecededByEscapeChar(escapeChar, match, offset, string) { return string[offset - 1] === escapeChar ? match : escapeChar + match;}
 	return str.replace(/(?:\\*)?"([^"\\]*\\.)*[^"]*"/g, function (match) {
 		return match
 			.replace(/\\"/g, '"')            // unescape double-quotes
-			.replace(/([^\\])'/g, '$1\\\'')  // escape single-quotes
+			.replace(/'/g, escapeIfNotPrecededByEscapeChar.bind(null, '\\'))
```

Comment:
@guyrotem can you update your PR and squash and we'll see if @sindresorhus wants to merge it?